src/transformers/models/gptsan_japanese/modeling_gptsan_japanese.py

```diff
@@ -0,0 +1,1326 @@
+            )
+        del gather_position
+
+        # Prepare head mask if needed
+        if head_mask is not None:
+            head_mask = self.get_head_mask(head_mask, self.config.n_layer)  # n_layer x batch x n_heads x N x N
+
+        present_key_value_states = tuple() if self.config.use_cache or use_cache else None
+        out_hidden_states = (hidden_states,) if self.config.output_hidden_states or output_hidden_states else None
+        all_attentions = tuple() if self.config.output_attentions or output_attentions else None
+        all_router_probs = tuple() if self.config.output_router_logits or output_router_logits else None
+
+        for layer, past in enumerate(pasts_or_squad_value):
+            if layer == self.config.num_switch_layers:
+                hidden_states = hidden_states + extra_position_embedding
+
+            output_router_tuple = (
+                self.config.output_router_logits or output_router_logits
+            ) and layer < self.config.num_switch_layers
+            block_output = self.blocks[layer](
+                hidden_states=hidden_states,
+                past_key_value=past,
+                attention_mask=attention_mask,
+                head_mask=head_mask,
+                use_cache=self.config.use_cache or use_cache,
+                output_attentions=self.config.output_attentions or output_attentions,
+                output_router_tuple=output_router_tuple,
+            )
+
+            outpos = 0
+            hidden_states = block_output[outpos]
+            if self.config.output_hidden_states or output_hidden_states:
+                out_hidden_states += (hidden_states,)
+            if self.config.use_cache or use_cache:
+                outpos += 1
+                present = block_output[outpos]
+                present_key_value_states += (present,)
+            if self.config.output_attentions or output_attentions:
+                outpos += 1
+                attention_probs = block_output[outpos]
+                all_attentions += (attention_probs,)
+            if output_router_tuple:
+                outpos += 1
+                router_tuple = block_output[outpos]
+                all_router_probs.append(router_tuple[0])
+
+        hidden_states = self.last_project(hidden_states)
+        hidden_states = self.act(hidden_states)
+
+        ret = collections.OrderedDict()
+        ret["last_hidden_state"] = hidden_states
+        if self.config.use_cache or use_cache:
+            ret["past_key_values"] = present_key_value_states
+        if self.config.output_hidden_states or output_hidden_states:
+            ret["hidden_states"] = out_hidden_states
+        if self.config.output_attentions or output_attentions:
+            ret["attentions"] = all_attentions
+        if self.config.output_router_logits or output_router_logits:
+            ret["router_probs"] = all_router_probs
+
+        if not return_dict:
+            return tuple(list(ret.values()))
```

Comment:
Reiterating here, let's juste use the usual notations.
```suggestion
        if output_hidden_states:
            all_hidden_states = all_hidden_states + (hidden_states,)

        if not return_dict:
            return tuple(
                v
                for v in [
                    hidden_states,
                    present_key_value_states,
                    all_hidden_states,
                    all_attentions,
                    all_cross_attentions,
                ]
                if v is not None
            )
        return BaseModelOutputWithPastAndCrossAttentions(
            last_hidden_state=hidden_states,
            past_key_values=present_key_value_states,
            hidden_states=all_hidden_states,
            attentions=all_attentions,
            cross_attentions=all_cross_attentions,
        )
```